consul/src/main/java/com/linecorp/armeria/internal/consul/ConsulClient.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.consul;
+
+import java.net.URI;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.linecorp.armeria.client.ClientDecoration;
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.WebClientBuilder;
+import com.linecorp.armeria.client.retry.RetryRule;
+import com.linecorp.armeria.client.retry.RetryingClient;
+import com.linecorp.armeria.common.HttpResponse;
+
+/**
+ * A client for accessing to Consul agent API server.
+ */
+public final class ConsulClient {
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private static final ClientOptions retryingClientOptions =
+            ClientOptions.of(ClientOptions.DECORATION.newValue(ClientDecoration.of(
+                    RetryingClient.newDecorator(RetryRule.failsafe(), 3))));
+
+    private final WebClient webClient;
+    private final AgentServiceClient agentClient;
+    private final CatalogClient catalogClient;
+    private final HealthClient healthClient;
+
+    public static ConsulClientBuilder builder() {
+        return new ConsulClientBuilder();
+    }
+
+    ConsulClient(URI uri, @Nullable String token) {
+        final WebClientBuilder builder = WebClient.builder(uri);
+        builder.options(retryingClientOptions);
+        if (token != null) {
+            // TODO(eugene70) test with token
+            builder.addHeader("X-Consul-Token", token);
```

Comment:
Could use `HttpHeaderNames.of("x-consul-token")` instead. Should keep it as a static final field.

Reply:
Yes.